pkg/reconciler/v1alpha1/eventlistener/eventlistener.go

```diff
@@ -61,12 +61,15 @@ const (
 )
 
 var (
-	// The container that we use to run in the EventListener Pods
-	elImage = flag.String("el-image", "override-with-el:latest",
+	// ELImage defines the container that we use to run in the EventListener Pods
+	ELImage = flag.String("el-image", "override-with-el:latest",
 		"The container image for the EventListener Pod.")
 	// ElPort defines the port for the EventListener to listen on
 	ElPort = flag.Int("el-port", 8080,
 		"The container port for the EventListener to listen on.")
+	// ELSetSecurityContext defines if the security context is set.
+	ELSetSecurityContext = flag.Bool("el-security-context", true,
```

Comment:
I would like to unit test this but have no idea how. Are there any examples where it's easy to tell how flags can be invoked for unit tests in this project?

Reply:
Current idea for me is to re-factor the flag setup in this controller to allow unit testing different values.

Reply:
One way might be to refactor one of the functions e.g. reconcileDeployment (or getDeployment) to take in a parameter called `isSecurityContexEnabled` and then we write unit tests for those functions (we lack those smaller unit tests today but maybe we should add them)

Another way might be to add these as fields in the Reconciler struct and we can set the right values in the test Reconcilers.

Reply:
> Another way might be to add these as fields in the Reconciler struct and we can set the right values in the test Reconcilers.

This is what I want to go for currently. I would like to do this as a follow up to this PR to prevent this from getting too big.

Reply:
sounds good to me!